packages/node_modules/@node-red/nodes/core/common/21-debug.js

```diff
@@ -108,43 +117,61 @@ module.exports = function(RED) {
             }
         })
         this.on("input", function(msg, send, done) {
-            if (hasOwnProperty.call(msg, "status") && msg.status &&
-                hasOwnProperty.call(msg.status, "source") && msg.status.source &&
-                hasOwnProperty.call(msg.status.source, "id") && (msg.status.source.id === node.id)) {
+            if (hasOwnProperty.call(msg, "status") && hasOwnProperty.call(msg.status, "source") && hasOwnProperty.call(msg.status.source, "id") && (msg.status.source.id === node.id)) {
                 done();
                 return;
             }
             if (node.tostatus === true) {
                 prepareStatus(msg, function(err,debugMsg) {
                     if (err) { node.error(err); return; }
-                    var output = debugMsg.msg;
-                    var st = (typeof output === 'string') ? output : util.inspect(output);
-                    var fill = "grey";
-                    var shape = "dot";
-                    if (typeof output === 'object' && hasOwnProperty.call(output, "fill") && hasOwnProperty.call(output, "shape") && hasOwnProperty.call(output, "text")) {
-                        fill = output.fill;
-                        shape = output.shape;
-                        st = output.text;
-                    }
-                    if (node.statusType === "auto") {
-                        if (hasOwnProperty.call(msg, "error")) {
-                            fill = "red";
-                            st = msg.error.message;
-                        }
-                        if (hasOwnProperty.call(msg, "status") &&
-                           msg.status) {
-                            fill = msg.status.fill || "grey";
-                            shape = msg.status.shape || "ring";
-                            st = msg.status.text || "";
+
+					if ( node.statusType === "counter" ){
```

Comment:
It would be better to handle this case outside of the call to `prepareStatus` - as we don't need to prepare anything in this scenario.

Reply:
Another piece of feedback for future contributions - I also see all of your indentation has been done with tabs - which doesn't match the rest of the file where we use spaces. Please take care to match the formatting of the files you are editing.